backend/btrixcloud/operator.py

```diff
@@ -498,6 +501,23 @@ async def mark_finished(
         """mark crawl as finished, set finished timestamp and final state"""
         finished = dt_now()
 
+        status.state = state
+        status.finished = to_k8s_date(finished)
+
+        if crawl:
+            await self.inc_crawl_complete_stats(crawl, finished)
+
+        asyncio.create_task(
+            self.do_crawl_finished_tasks(redis, crawl_id, cid, state, stats, finished)
+        )
+
+        return status
+
+    # pylint: disable=too-many-arguments
+    async def do_crawl_finished_tasks(
+        self, redis, crawl_id, cid, state, stats, finished
+    ):
+        """Run tasks after crawl completes in asyncio.task coroutine."""
         kwargs = {"state": state, "finished": finished}
```

Comment:
This part / `update_crawl` can probably be moved back into mark_finished(), since it just updates the db with one setting.

The rest are potentially longer running and should stay here..